client/lib/route/path.js

```diff
@@ -27,6 +27,20 @@ export function getSiteFragment( path ) {
 	// There are 2 URL positions where we should look for the site fragment:
 	// last (most sections) and second-to-last (post ID is last in editor)
 
+	// Though, in some`/me/purchases` paths, it could also be in third position,
+	// right after the `/me/purchases/` part.
+	// e.g. /me/purchases/example.wordpress.com/foo/bar
+	if ( 0 === basePath.indexOf( '/me/purchases/' ) ) {
```

Comment:
Could we optionally pass in the position in the function signature? For example:
```jsx
export function getSiteFragment( path, sitePosition = null ) {
    if( sitePosition ) {
        const siteFragment = pieces[ sitePosition ];
        //...
}
// in your new tests:
`getSiteFragment( path, 3 );`
```

Reply:
I think this would complicate its usage.
Think of `PageViewTracker`, which is what spurred this change: we'd basically move the `/me/purchases/` check there instead of here. E.g.

```es6
const currentSlug = getSiteFragment(
  window.location.pathname,
  0 === basePath.indexOf( '/me/purchases/' ) ? 3 : null
);
```